app/models/slack_bot.rb

```diff
@@ -3,9 +3,13 @@ class SlackBot < ActiveRecord::Base
 
   def self.from_auth_hash(auth_hash)
     bot_info = auth_hash[:extra][:bot_info]
-    find_or_create_by(id: bot_info[:bot_user_id]) do |bot|
+    bot = find_or_create_by(id: bot_info[:bot_user_id]) do |bot|
```

Comment:
This all might read a little better if it was just turned into:

```ruby
bot = find_by(id: bot_info[:bot_user_id])
if bot
  bot.update_attributes(...)
else
  create(...)
end
```

Reply:
fixed I think.